terraform/services/snyk-integration/main.tf

```
@@ -0,0 +1,67 @@
+data "aws_iam_policy" "developer_boundary_policy" {
+  name = "developer-boundary-policy"
+}
+
+data "aws_ssm_parameter" "external_id" {
+  name = "/snyk-integration/external_id"
+}
+
+data "aws_ssm_parameter" "ecr_integration_user" {
+  name = "/snyk-integration/ecr_integration_user"
+}
+
+data "aws_iam_policy_document" "snyk_trust" {
+  statement {
+    effect = "Allow"
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_ssm_parameter.ecr_integration_user.value]
+    }
+    actions = ["sts:AssumeRole"]
+    condition {
+      test     = "StringEquals"
+      variable = "sts:ExternalId"
+      values   = [data.aws_ssm_parameter.external_id.value]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "snyk_pull" {
+  statement {
+    sid    = "SnykAllowPull"
+    effect = "Allow"
+    actions = [
+      "ecr:GetLifecyclePolicyPreview",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      "ecr:DescribeImages",
+      "ecr:GetAuthorizationToken",
+      "ecr:DescribeRepositories",
+      "ecr:ListTagsForResource",
+      "ecr:ListImages",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetRepositoryPolicy",
+      "ecr:GetLifecyclePolicy"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "snyk_pull" {
+  name        = "${var.app}-${var.env}-snyk-pull"
+  path        = "/delegatedadmin/developer/"
+  description = "Policy for Snyk to pull images from ECR"
+  policy      = data.aws_iam_policy_document.snyk_pull.json
+}
+
+resource "aws_iam_role" "snyk" {
+  name                 = "${var.app}-${var.env}-snyk"
```

Since we'll have just one snyk role per account, let's drop the app and env from the name

```suggestion
  name                 = "snyk"
```